dev/benchmarks/microbenchmarks/lib/geometry/matrix_utils_transform_rect_bench.dart

```diff
@@ -0,0 +1,139 @@
+    Matrix4.identity()..scale(1.2, 1.3, 1.0)..translate(12.0, 13.0, 10.0),
+  ];
+  final List<Matrix4> _perspectiveTransforms = <Matrix4>[
+    _makePerspective(10.0, math.pi / 8.0, 0.3),
+    _makePerspective( 8.0, math.pi / 8.0, 0.2),
+    _makePerspective( 1.0, math.pi / 4.0, 0.1)..rotateX(0.1),
+  ];
+  final List<Rect> _rects = <Rect>[
+    const Rect.fromLTRB(1.1, 1.2, 1.5, 1.8),
+    const Rect.fromLTRB(1.1, 1.2, 0.0, 1.0),
+    const Rect.fromLTRB(1.1, 1.2, 1.3, 1.0),
+    const Rect.fromLTRB(-1.1, -1.2, 0.0, 1.0),
+    const Rect.fromLTRB(-1.1, -1.2, -1.5, -1.8),
+  ];
+  final List<Offset> _offsets = <Offset>[
+    const Offset(1.1, 1.2),
+    const Offset(1.5, 1.8),
+    const Offset(0.0, 0.0),
+    const Offset(-1.1, -1.2),
+    const Offset(-1.5, -1.8),
+  ];
+
+  // Warm up lap
+  for (int i = 0; i < _kNumWarmUp; i += 1) {
+    final Rect rect = _rects[i % _rects.length];
+    final Offset offset = _offsets[i % _offsets.length];
+    final Matrix4 transform = (i > _kNumWarmUp / 2)
+        ? _perspectiveTransforms[i % _perspectiveTransforms.length]
+        : _transforms[i % _transforms.length];
+    MatrixUtils.transformRect(transform, rect);
+    MatrixUtils.transformPoint(transform, offset);
+  }
+
+  final Stopwatch watch = Stopwatch();
+  watch.start();
+  for (int i = 0; i < _kNumIterations; i += 1) {
+    final Rect rect = _rects[i % _rects.length];
+    final Matrix4 transform = _perspectiveTransforms[i % _perspectiveTransforms.length];
+    MatrixUtils.transformRect(transform, rect);
+  }
+  watch.stop();
+  final int rectMicrosecondsPerspective = watch.elapsedMicroseconds;
+
+  watch.reset();
+  watch.start();
+  for (int i = 0; i < _kNumIterations; i += 1) {
+    final Rect rect = _rects[i % _rects.length];
+    final Matrix4 transform = _transforms[i % _transforms.length];
+    MatrixUtils.transformRect(transform, rect);
+  }
+  watch.stop();
+  final int rectMicrosecondsAffine = watch.elapsedMicroseconds;
+
+  watch.reset();
+  watch.start();
+  for (int i = 0; i < _kNumIterations; i += 1) {
+    final Offset offset = _offsets[i % _offsets.length];
+    final Matrix4 transform = _perspectiveTransforms[i % _perspectiveTransforms.length];
+    MatrixUtils.transformPoint(transform, offset);
+  }
+  watch.stop();
+  final int pointMicrosecondsPerspective = watch.elapsedMicroseconds;
+
+  watch.reset();
+  watch.start();
+  for (int i = 0; i < _kNumIterations; i += 1) {
+    final Offset offset = _offsets[i % _offsets.length];
+    final Matrix4 transform = _transforms[i % _transforms.length];
+    MatrixUtils.transformPoint(transform, offset);
+  }
+  watch.stop();
+  final int pointMicrosecondsAffine = watch.elapsedMicroseconds;
+
+  final BenchmarkResultPrinter printer = BenchmarkResultPrinter();
+  const double scale = 1000.0 / _kNumIterations;
+  printer.addResult(
+    description: 'MatrixUtils.transformRectPerspective',
+    value: rectMicrosecondsPerspective * scale,
+    unit: 'ns per iteration',
```

Comment:
This unit might be too long for our dashboard to display properly. Maybe just set the unit to `ns`, and set the name to `..._time_per_iteration`

Reply:
Looking at the dashboard on my laptop screen it looks like there is plenty of room, but the prevailing standard seems to be to just list time units and not "per <anything>". I think "per iteration" can be implied?

Reply:
Hmmm.  In looking at the other microbenchmarks, they all have "*s per iteration" as their units. I think it would be better to match the other microbenchmarks, no?

Reply:
Even stranger - if you look at the rrect benchmark in the same directory it also uses the "per iteration" description of the units, but if you look at its output on the dashboard, it just says "ms". Whatever publishes the results on the dashboard does its own captioning. I haven't found that code yet but I want to so I can see if it will convert these results to "ms" which would make them so tiny as to be unreadable.  I'm hoping it auto-scales the units and can handle operations that take only a few nanos.

Reply:
Oh, I didn't know that our dashboard automatically strips out the unit. Then this looks good to me. I think it doesn't auto-scales the units because sometimes I see 10000K ns.